taca/analysis/analysis.py

```diff
@@ -4,319 +4,214 @@
-    :param str qc_file: Path to file with information about transferred runs
-    """
-    already_seen=False
-    runname=os.path.basename(os.path.abspath(run))
-    shortrun=runname.split('_')[0] + '_' +runname.split('_')[-1]
-    with open(qc_file, 'ab+') as f:
-        f.seek(0)
-        for row in f:
-            #Rows have two columns: run and transfer date
-            if row.split('\t')[0] == runname: 
-                already_seen=True
-
-        if not already_seen:
-            if status:
-                f.write("{}\tPASSED\n".format(runname))
-            else:
-                sj="{} failed QC".format(runname)
-                cnt="""The run {run} has failed qc and will NOT be transfered to Nestor.
-
-                       The run might be available at : https://genomics-status.scilifelab.se/flowcells/{shortfc}
-                       
-                       To read the logs, run the following command on {server} 
-                       grep -A30 "Checking run {run}" {log}
-                       
-                       To force the transfer : 
-                        taca analysis transfer {rundir} """.format(run=runname, shortfc=shortrun, log=CONFIG['log']['file'], server=os.uname()[1], rundir=run)
-                rcp=CONFIG['mail']['recipients']
-                misc.send_mail(sj, cnt, rcp)
-                f.write("{}\tFAILED\n".format(os.path.basename(run)))
-
 def upload_to_statusdb(run_dir):
     """
+        interface for click
+    """
+    sequencer_type = _run_type(run_dir)
+    if sequencer_type is 'HiSeqX':
+        runObj = HiSeqX_Run(run_dir, CONFIG["analysis"]["HiSeqX"])
+    elif sequencer_type is 'HiSeq':
+        print "not yet implemented: HiSeq"
+        return
+    elif sequencer_type is 'MiSeq':
+        print "not yet implemented: miseq"
+        return
+    _upload_to_statusdb(runObj)
+
+
+def _upload_to_statusdb(run):
+    """
     Triggers the upload to statusdb using the dependency flowcell_parser
-    
-     :param string run_dir: the run directory to upload
+    :param Run run: the object run
     """
     couch = fcpdb.setupServer(CONFIG)
-    db=couch[CONFIG['statusdb']['xten_db']]
-    parser=XTenParser(run_dir)
-    fcpdb.update_doc(db,parser.obj)
+    db    = couch[CONFIG['statusdb']['xten_db']]
+    parser = run.runParserObj
+    fcpdb.update_doc( db , parser.obj)
 
-def run_preprocessing(run):
+
+def transfer_run(run_dir, analysis):
+    """
+    interface for click
+    :param: string run_dir: the run to tranfer
+    :param bool analysis: if trigger or not the analysis
+    """
+    sequencer_type = _run_type(run_dir)
+    if sequencer_type is 'HiSeqX':
+        runObj = HiSeqX_Run(run_dir, CONFIG["analysis"]["HiSeqX"])
+    elif sequencer_type is 'HiSeq':
+        print "not yet implemented: HiSeq"
+        return
+    elif sequencer_type is 'MiSeq':
+        print "not yet implemented: miseq"
+        return
+    _transfer_run(runObj, analysis)
+
+
+def _transfer_run(run, analysis):
```

Comment:
thanks @senthil10 you are perfectly right, I removed the _tranfer_run as it was useless